Assignment/Conversion.c

```diff
@@ -41,35 +47,42 @@ void DecToBinary (int n) {
             break;
          }
       }
-      // If there is still a carry left, extend the binary array
-      if (carry == 1) {
-         binary[i] = 1;
-         i++;
+   }
+   else {
+      bitLength = (i <= 8) ? 8 : i;
+      for (int j = i; j < bitLength; j++) {
+         binary[j] = 0;
       }
+      i = bitLength;
    }
-   for (int k = i - 1; k >= 0; k--)    printf ("%d", binary[k]);   
+   for (int k = i - 1; k >= 0; k--)   printf ("%d", binary[k]);   
 }
 
-// Function to convert the decimal number to Hexadecimal
+/// <summary>Decimal number to Hexadecimal Conversion</summary>
 void DecToHexa (int n) {
    char hexa[32];
    int i = 0;
-   if (n == 0)  printf ("Hexadecimal is 0");
+   if (n == 0)   printf ("0");  
    unsigned int num = (unsigned int)n;   // Handle negative numbers by using unsigned int
    while (num > 0) {
       int rem = num % 16;
       if (rem < 10)  hexa[i] = rem + '0';
       else  hexa[i] = rem - 10 + 'A';

```

Comment:
Can rewrite this if else statement using ternary operator.
[https://www.geeksforgeeks.org/ternary-operator-in-programming/](url)